### PR TITLE
[Finder] Add .gitignore nested negated patterns support

### PR DESCRIPTION
**Pull request by julienfalque**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Extracted from #43239.

### CODE REVIEW

File: src/Symfony/Component/Finder/Iterator/VcsIgnoredFilterIterator.php [hunk 1]
```diff
@@ -37,25 +42,52 @@ public function accept(): bool
...
+            }
+
             $fileRelativePath = substr($fileRealPath, \strlen($parentDirectory) + 1);
 
-            $regex = $this->readGitignoreFile("{$parentDirectory}/.gitignore");
+            $regexps = $this->readGitignoreFile("{$parentDirectory}/.gitignore");
 
-            if (null !== $regex && preg_match($regex, $fileRelativePath)) {
-                return false;
+            if (null === $regexps) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Finder/Iterator/VcsIgnoredFilterIterator.php]: can be merged with the previous line `if (null === $regexps = ...`

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Finder/Iterator/VcsIgnoredFilterIterator.php (hunk 1 above)
Comment (julienfalque) [reply src/Symfony/Component/Finder/Iterator/VcsIgnoredFilterIterator.php]: Done.

**Comment by nicolas-grekas** [comment]
Thank you @julienfalque.

**Comment by julienfalque** [comment]
Thank you @nicolas-grekas!

**Pull request merged**